testsuites/syncgateway/functional/tests/test_log_redaction.py

```diff
@@ -0,0 +1,454 @@
+
+    # 3. Create user in sync_gateway
+    sg_client = MobileRestClient()
+    channels = ["log-redaction"]
+    sg_client.create_user(url=sg_admin_url, db=sg_db, name='autotest', password='validkey', channels=channels)
+    autouser_session = sg_client.create_session(url=sg_admin_url, db=sg_db, name='autotest', password='validkey')
+
+    # 4. Create docs with xattrs
+    sgdoc_bodies = document.create_docs(doc_id_prefix='sg_docs', number=num_of_docs,
+                                        attachments_generator=attachment.generate_2_png_10_10, channels=channels)
+    sg_client.add_bulk_docs(url=sg_url, db=sg_db, docs=sgdoc_bodies, auth=autouser_session)
+    assert len(sgdoc_bodies) == num_of_docs
+
+    # should throw an error when trying with upload host without upload
+    upload_host = "https://s3.amazonaws.com/cb-customers"
+    resp = sg_client.sgCollect_restCall(sg_host, redact_level="partial", upload_host=upload_host)
+    assert "Invalid options used for sgcollect_info: upload must be set to true if upload_host is specified" in resp.content
+    customer = "customer-name"
+    ticket = "123"
+
+    # should throw an error when trying with customer without upload parameter
+    resp = sg_client.sgCollect_restCall(sg_host, redact_level="partial", customer=customer)
+    assert "Invalid options used for sgcollect_info: upload must be set to true if customer is specified" in resp.content
+
+    # should throw an error when trying ticket without upload
+    resp = sg_client.sgCollect_restCall(sg_host, redact_level="partial", ticket=ticket)
+    assert "Invalid options used for sgcollect_info: upload must be set to true if ticket is specified" in resp.content
+
+    # should throw an error when trying with upload_host, customer and ticket without upload
+    resp = sg_client.sgCollect_restCall(sg_host, redact_level="partial", upload_host=upload_host, customer=customer, ticket=ticket)
+    assert "Invalid options used for sgcollect_info: upload must be set to true if upload_host is specified" in resp.content
+
+    # should throw an error when trying with output dir which does not exist
+    output_dir = "/abc"
+    resp = sg_client.sgCollect_restCall(sg_host, redact_level="partial", output_directory=output_dir)
+
+    assert "Invalid options used for sgcollect_info: no such file or directory:" in resp.content
+
+
+def verify_log_redaction(cluster_config, log_redaction_level, mode):
+    ansible_runner = AnsibleRunner(cluster_config)
+
+    log_info("Pulling sync_gateway / sg_accel logs")
+    # fetch logs from sync_gateway instances
+    status = ansible_runner.run_ansible_playbook("fetch-sync-gateway-all-logs.yml")
+    if status != 0:
+        raise CollectionError("Could not pull logs")
+    temp_log_path = ""
+    # zip logs and timestamp
+    if os.path.isdir("/tmp/sg_logs"):
+        date_time = time.strftime("%Y-%m-%d-%H-%M-%S")
+        temp_log_path = "/tmp/{}-{}-sglogs".format("log-redaction", date_time)
```

Comment:
I am curious, why we are creating logs under /tmp/sg_logs and then moving it to /tmp/log-redaction-timestamp. Can't we name the zip file as log-redaction-timestamp.zip instead of copying the files around?

Reply:
I replaced fetch-sync-gateway-logs.yml with fetch-sync-gateway-all-logs.yml with you earlier comment , which was already there. That has step to copy to sg_logs , that is why I need to copy to /tmp/log-redaction-timestamp